scanner/templates/node/Dockerfile

```
@@ -1,38 +1,53 @@
-FROM debian:bullseye as builder
+# syntax = docker/dockerfile:1
 
-ENV PATH=/usr/local/node/bin:$PATH
 ARG NODE_VERSION={{ .nodeVersion }}
+FROM node:${NODE_VERSION}-slim as base
+
+LABEL fly_launch_runtime="{{ .runtime }}"
+
+ENV NODE_ENV production
+
 {{ if .yarn -}}
 ARG YARN_VERSION={{ .yarnVersion }}
-{{ end }}
-RUN apt-get update; apt install -y curl python-is-python3 pkg-config build-essential && \
-    curl -sL https://github.com/nodenv/node-build/archive/master.tar.gz | tar xz -C /tmp/ && \
-    /tmp/node-build-master/bin/node-build "${NODE_VERSION}" /usr/local/node && \
-{{ if .yarn -}}
-    npm install -g yarn@$YARN_VERSION && \
-{{ end -}}
-    rm -rf /tmp/node-build-master
+RUN npm install -g yarn@$YARN_VERSION
 
+{{ end -}}
 RUN mkdir /app
 WORKDIR /app
 
-COPY . .
 
+FROM base as build
+
+RUN apt-get update; && \
+    apt install -y python-is-python3 pkg-config build-essential {{ 
+    if .prisma }}openssl {{ end }}
+
+COPY {{ .package_files }} .
+RUN {{ .packager }} install --production=false
+
+{{ if .prisma -}}
+ADD prisma .
+RUN npx prisma generate
+
+{{ end -}}
+COPY . .
 {{ if .build -}}
-RUN {{ .packager }} install && {{ .packager }} run build
-{{ else -}}
-RUN {{ .packager }} install
+    
+RUN {{ .packager }} run build
 {{ end }}
 
-FROM debian:bullseye-slim
-
-LABEL fly_launch_runtime="nodejs"
+FROM base
 
-COPY --from=builder /usr/local/node /usr/local/node
 COPY --from=builder /app /app
```

```suggestion
COPY --from=build /app /app
```

Not sure if we should add everything though 🤔
As explained in https://github.com/remix-run/indie-stack/pull/90, this would include _all_ files available in the repo

The current `Dockerfile` in our `indie-stack` does the following:

```Dockerfile
COPY --from=production-deps /myapp/node_modules /myapp/node_modules
COPY --from=build /myapp/node_modules/.prisma /myapp/node_modules/.prisma

COPY --from=build /myapp/build /myapp/build
COPY --from=build /myapp/public /myapp/public
COPY --from=build /myapp/package.json /myapp/package.json
COPY --from=build /myapp/start.sh /myapp/start.sh
COPY --from=build /myapp/prisma /myapp/prisma
```